graphql_compiler/global_utils.py

```diff
@@ -0,0 +1,11 @@
+# Copyright 2019-present Kensho Technologies, LLC.
+def partition(iterable, pred):
```

Comment:
I believe we already have `funcy` as a dependency, let's use it. Be careful to use it in a python-version-agnostic way, it does some strange things if you don't specify py2 vs py3 semantics.

https://funcy.readthedocs.io/en/stable/seqs.html#split